### PR TITLE
Added opt argument for js2r-forward-slurp/barf.

### LINKED ISSUE #18: Optional number argument for js2r-forward-slurp / barf
Feature creep request :)
Would be really nice to be able to do "M-3 C-c C-m s l" instead of:
"C-c C-m s l" followed by "C-x z z z". It would work similar to how kill-line and delete-char works.

### PR DESCRIPTION
**Pull request by gregersson**
Supports positive prefix arguments.
Uses simple loop function from cl.el (already required in js2-mode.el)
Fixes #18

### CODE REVIEW

**Comment by gregersson** [comment]
@NicolasPetton Hi! I have finally removed the loop construct. Let me know if you have any further feedback on this pull request. Cheers

**Comment by NicolasPetton** [comment]
@gregersson cool that you have written tests :)
There are small issues regarding formatting in some functions, could you fix that?

**Comment by gregersson** [comment]
@NicolasPetton Sure, I found one confusing indentation error that I have fixed now. Did you have any more specific issues in mind?

File: js2r-paredit.el [hunk 1]
```diff
@@ -139,7 +139,18 @@ When at the beginning of the node, kill from outside of it."
...
          (next-sibling (js2-node-next-sibling standalone))
          (beg (js2-node-abs-pos next-sibling))
-         (end (1+ (js2-node-abs-end next-sibling))) ;; include whitespace after statement
+         (last-sibling (if (wholenump arg)
+                           (let ((num arg)
+                                 (iter-sibling next-sibling))
+                             (while (> num 1) ;; Do next-sibling arg nbr of times
+                               (setq iter-sibling (js2-node-next-sibling iter-sibling))
+                               (setq num (1- num))
+                               )
```
Comment (NicolasPetton) [inline js2r-paredit.el]: Parenthesis should be closed on the same :)

File: js2r-paredit.el [hunk 2]
```diff
@@ -161,8 +172,19 @@ When at the beginning of the node, kill from outside of it."
...
+                               (let ((num arg)
+                                     (iter-child last-child))
+                                 (while (> num 1) ;; Do prev-sibling arg nbr of times
+                                   (setq iter-child (js2-node-prev-sibling iter-child))
+                                   (setq num (1- num))
+                                   )
+                                 iter-child
+                                 )
+                             last-child);; No optional arg. Just use last-child
+                           )
```
Comment (NicolasPetton) [inline js2r-paredit.el]: Same here

**Comment by NicolasPetton** [comment]
@gregersson I commented the PR

**Comment by gregersson** [comment]
@NicolasPetton Ah, I see! Thank you for the pointers. I have uploaded a new changeset that looks a lot better. :)

**Comment by NicolasPetton** [comment]
Thanks, I'll have a look asap!

Pär Gregersson <notifications@github.com> writes:

> @NicolasPetton Ah, I see! Thank you for the pointers. I have uploaded a new changeset that looks a lot better. :)
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/magnars/js2-refactor.el/pull/19#issuecomment-72455621

-- 
Nicolas Petton
http://nicolas-petton.fr

**Comment by NicolasPetton** [comment]
Sorry for the looooong delay. LGTM!

**Pull request merged**